source/brailleDisplayDrivers/alva.py

```diff
@@ -258,44 +272,59 @@ def _handleInput(self, group, number):
 			# This begins a new key combination.
 			self._ignoreKeyReleases = False
 
-	def _hidDisplay(self, cells):
-		for offset in range(0, len(cells), ALVA_BRAILLE_OUTPUT_MAX_SIZE):
-			cellsToWrite = cells[offset:offset+ALVA_BRAILLE_OUTPUT_MAX_SIZE]
-			self._dev.write("{id}{offset}{count}{cells}".format(
-				id=ALVA_BRAILLE_OUTPUT_REPORT,
-				offset=chr(offset),
-				count=chr(len(cellsToWrite)),
-				cells=cellsToWrite
-			))
-
-	def _ser6Display(self, cells):
-		self._ser6SendMessage(b"B", chr(0)+chr(len(cells))+cells)
-
-	def display(self, cells):
+	def _hidDisplay(self, cellBytes: bytes) -> None:
+		for offset in range(0, len(cellBytes), ALVA_BRAILLE_OUTPUT_MAX_SIZE):
+			cellsToWrite = cellBytes[offset:offset+ALVA_BRAILLE_OUTPUT_MAX_SIZE]
+			data = b"".join([
+				ALVA_BRAILLE_OUTPUT_REPORT,
+				intToByte(offset),
+				intToByte(len(cellsToWrite)),
+				cellsToWrite
+			])
+			self._dev.write(data)
+
+	def _ser6Display(self, cellBytes: bytes) -> None:
+		if not isinstance(cellBytes, bytes):
+			raise TypeError("Expected param 'cells' to be of type 'bytes'")
+		value = b"".join([
+			b"\x00",
+			intToByte(len(cellBytes)),
+			cellBytes
+		])
+		self._ser6SendMessage(b"B", value)
+
+	def display(self, cells: List[int]):

```

Comment:
I think now we can, we should consider making this a bytes or bytearray object. In most if not all cases, one of the first things that happens in this method is converting a list to a byte sequence.

Reply:
I understand, however, I think it's best to start with the smallest change set. This isn't really a change from how it worked with python 2, and it because it is quite self contained it should be fairly easy to do after this PR.